### PR TITLE
Ad/compat linting

### PR DESCRIPTION
**Pull request by adaddeo**
This introduces a browserslist configuration file. This will be used by babel plugins to decide what transformations need to be done (and thus reducing bundle size the less browsers need to be targetd) and also by the eslint plugin I added to warn you about features you are using that aren't supported by targeted browsers in your config.

I'd also recommend adding a section to the readme letting users know they need to polyfill Promise if they want it to work in IE.

### CODE REVIEW

**Comment by adaddeo** [comment]
Disclaimer: this might be a workable config but I kind of just threw it together as a quick proof-of-concept, but I'd validate it's doing what you want/the best way to do this.

**Comment by j-berman** [comment]
@adaddeo @dvassallo how's this look?

File: src/userbase-js/src/api/auth.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ const getServerPublicKey = async () => {
 
     const method = 'GET'
     const url = `${config.getEndpoint()}/api/auth/server-public-key`
-    const timeout = timeout
+    const timeout = TEN_SECONDS_MS
```
Comment (j-berman) [inline src/userbase-js/src/api/auth.js:132]: Just want to point this out. Haven't deeply investigated yet, but not sure why this didn't cause tests to fail earlier. Seems like there may be a caching issue going on both locally and with our CI tests.

**Comment by dvassallo** [comment]
Looks good to me!

**Pull request merged**